$NetBSD: patch-configure.ac,v 1.4 2015/08/31 15:26:19 sevan Exp $

* Detect NetBSD libatomic
* test(1) portability fix
* Add support for Bitrig

--- configure.ac.orig	Fri Mar 27 21:17:48 2015
+++ configure.ac
@@ -409,6 +409,18 @@ if test "x$ac_cv_header_elf_h" = "xyes"; then
     CONFIG_CPPFLAGS="$CONFIG_CPPFLAGS -DU_HAVE_ELF_H=1";
 fi
 
+# check if sys/atomic.h is present.
+AC_CHECK_HEADERS([sys/atomic.h])
+case "${host}" in
+*-netbsd*)
+if test "x$ac_cv_header_sys_atomic_h" = "xyes"; then
+    CONFIG_CPPFLAGS="$CONFIG_CPPFLAGS -DU_HAVE_NETBSD_ATOMIC_OPS=1";
+fi
+;;
+*)
+;;
+esac
+
 U_ENABLE_DYLOAD=1
 enable=yes
 AC_MSG_CHECKING([whether to enable dynamic loading of plugins])
@@ -1234,7 +1246,7 @@ ICUDATA_CHAR=$U_ENDIAN_CHAR
 case "${host}" in
 	*-*-solaris*) 	platform=U_SOLARIS ;;
 	*-*-linux*|*-*-gnu|*-*-k*bsd*-gnu|*-*-kopensolaris*-gnu)        platform=U_LINUX ;;
-	*-*-*bsd*|*-*-dragonfly*) 	platform=U_BSD ;;
+	*-*-*bsd*|*-*-dragonfly*|*-*-bitrig*) 	platform=U_BSD ;;
 	*-*-aix*) 	platform=U_AIX ;;
 	*-*-hpux*) 	platform=U_HPUX ;;
 	*-apple-darwin*|*-apple-rhapsody*)	platform=U_DARWIN ;;
@@ -1253,6 +1265,7 @@ case "${host}" in
 	*-*-beos)		platform=U_BEOS ;;
 	*-*-irix*)		platform=U_IRIX ;;
 	*-ncr-*)		platform=U_MPRAS ;;
+	*-*-sco3.2v5*)		platform=USCOOSR5 ;;
 	*) 		platform=U_UNKNOWN_PLATFORM ;;
 esac
 AC_SUBST(ICUDATA_CHAR)
@@ -1270,13 +1283,13 @@ fi
 # Now that we're done using CPPFLAGS etc. for tests, we can change it
 # for build.
 
-if test "${CC}" == "clang"; then
+if test "${CC}" = "clang"; then
    CLANGCFLAGS="-Qunused-arguments -Wno-parentheses-equality"
 else
    CLANGCFLAGS=""
 fi
 
-if test "${CXX}" == "clang++"; then
+if test "${CXX}" = "clang++"; then
    CLANGCXXFLAGS="-Qunused-arguments -Wno-parentheses-equality"
 else
    CLANGCXXFLAGS=""
